src/js/apps/patients/sidebar/action-sidebar_app.js

```diff
@@ -236,6 +236,7 @@ export default App.extend(extend({
     });
     attachment.upload(file);
 
+    this.action.addAttachment(attachment);
```

Comment:
When passing the attachment to the `addAttachment` function here, `attachment.id` is `undefined`. Which will cause the test I created in this PR to fail.

Reply:
I think we need to wait until the upload is finished.

We could do something like this:

```js
this.listenTo(attachment, 'upload:success', uploadedAttachment => {
  this.action.addAttachment(uploadedAttachment);
  Radio.request('ws', 'add', uploadedAttachment);
});
```

Or:

```js
attachment.upload(file).then(uploadedFile => {
  this.action.addAttachment(uploadedAttachment);
  Radio.request('ws', 'add', uploadedAttachment);
});

Reply:
And if an upload fails (`'upload:failed'` triggered), this would also stop the paperclip icon and websocket from being added when they shouldn't be.